### PR TITLE
added codecs

### PR DESCRIPTION
**Pull request by bilalyasar**

### CODE REVIEW

File: src/codec/ClientCreateProxyCodec.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+
+
+
+static calculateSize(name, serviceName, target){
+    // Calculates the request payload size
+    var dataSize = 0;
+    dataSize += BitsUtil.calculateSizeStr(name);
+    dataSize += BitsUtil.calculateSizeStr(serviceName);
+    dataSize += BitsUtil.calculateSizeCom.hazelcast.nio.address(target);
```
Comment (mustafaiman) [inline src/codec/ClientCreateProxyCodec.ts]: I guess this should be `BitsUtil.calculateSize(target).

File: src/codec/ClientCreateProxyCodec.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    return dataSize;
+}
+
+static encodeRequest(name, serviceName, target){
+    // Encode request into clientMessage
+    var payloadSize;
+    var clientMessage = new ClientMessage(payloadSize=this.calculateSize(name, serviceName, target));
```
Comment (mustafaiman) [inline src/codec/ClientCreateProxyCodec.ts]: There is no `ClientMessage(size: number)`. This should be `newClientMessage(...)`

**Comment by asimarslan** [comment]
@bilalyasar , This PR is very big and github cannot display it. Can you split it into some logical parts like Map, Queue List, Set. Can make sure we can review it here.

**Comment by asimarslan** [comment]
And do not add any codec that Python client do not have like Cache, MapReduce. Not all distributed object supported by non-java clients

**Comment by bilalyasar** [comment]
@asimarslan , ok will split them.

**Pull request closed without merging**